### DISCUSSION TITLE
Ability to add new fields to existing collection

### DISCUSSION THREAD

**Discussion by RithikRaj64** [milvus-io/milvus#35221] [Ideas & Feature requests]
- I am working on a project where I am required to add a new field to an already existing collection.
- The only workaround that I could find is to create a new collection and with the existing data which is not much efficient.
- It would be helpful, if we could dynamically add more fields to an existing collection.

**Comment by yhmo** [comment]
Currently, milvus doesn't support editing a collection's schema. It is a feature in plan, I suppose it will be available in a future major version (2.5 or 3.0).

In the current versions, if your collection has enabled dynamic schema, you can add new fields when you call insert().
https://milvus.io/docs/v2.3.x/dynamic_schema.md#Dynamic-Schema

**Reply by xiaofan-luan** [reply]
We are targeting to deliver this feature on 3.0
